src/lib/functionBodyParser.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ *
+ * Convert function to a string ang gets body expressions
+ * this expressions' strings will then be inserted in the worker blob
+ *
+ * @param {Function} [] A function whose will be extracted
+ * @returns {String} The function's body
+ *
+ * @example
+ * functionBodyParser(function () {let a = 2}) // return let a = 2)
+ */
+const functionBodyParser = (fn?: Function) => {
+  const functionString = fn?.toString()
+
+  return functionString?.slice(
+    functionString.indexOf('{') + 1,
+    functionString.lastIndexOf('}')
+  )
+}
```

Comment:
Have you ever considered to use a RegExp?

```javascript
const result = /^.*\{(.*)\}$/.exec(functionString)

return result ? result[1] : ''
```

This should do the trick :smile:

Reply:
Hey! I did tried regex but had a few issues so just went with the built-in JS character match for that one. Thanks for sharing this piece of code.

Unfortunately, I couldn’t quite make it work. I saved a snippet on RegExr [here](https://regexr.com/551pl) and as you can see it's currently working only for one-line declarations. And also the function just needs the body (without the brackets), not the whole declaration. Do you think you can make it work? I tried but I'm not that pro with regex 😅

Reply:
@gonzachr Try this one and let me know :+1: 

```javascript
const result = /^.*\{([^\{\}]*)\}/.exec(functionString)

return result ? result[1] : ''
```

Reply:
Looking better! Unfortunately I just need the _body_ of the function as this implementation need to evaluate not the function but the body of it, so the variables declared in the function are available in the global worker scope. If we were to for example use IIFE, the scope would not be the global worker scope :(

This also raises a thought that I had the past days which is that is a bit weir to have global variables living around and using them in functions, feels a bit weird. In the sense that you have to explicitly know which variables you set, maybe we can be a bit more creative here. But for now I'll continue with the implementation and fix the conflicts cc @alewin

Reply:
Yes it's a bit weird, in fact the management of global variables, using the IIFE is much more complicated, perhaps to make this feature feasible, we should rewrite the `jobrunner` but for the moment I have no other ideas 🤔

Reply:
I agree, I think I rushed a bit to open this PR 😅it was just fun to implement at the time. Do you think we should spend more time thinking about this one? Or is it good to go?

Reply:
yes perhaps it is better to think of an alternative solution, thanks anyway @gonzachr  :)

Reply:
Great, I like that better! I'm closing this and we can maybe turn it into a proposal